packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -0,0 +1,11 @@
+import 'dart:ui';
+
+import '../../../components.dart';
+import '../../palette.dart';
+
+mixin HasPaint on BaseComponent {
```

Comment:
Is it possible for this to have a list of paints (or a map) maybe? I'm thinking that many components might have more than one paint.
And then we could specify with index / map key which of them that should be changed in the effect?

Reply:
Hmmm, it is possible sure, I can't think on components that may need more than one paint, do you have an example?

My fear of needing to pass a name or index to the effects API is that the effects api may end up a little confusing

Reply:
Anything with a background and a foreground, but I agree it might become confusing.
But the constructor could default to the first if there is no map-key specified? If it most often would be just one.

Reply:
@spydon I still need to improve docs and make tests for this but I have applied the changes to enable this suggestion, lmk what do you think.